src/Microsoft.AspNetCore.Mvc.Core/project.json

```diff
@@ -40,6 +40,7 @@
       "type": "build",
       "version": "1.1.0-*"
     },
+    "Microsoft.Extensions.Logging": "1.1.0-preview2-22526",
```

Comment:
Ah, got it. Issue here is that `BodyModelBinder` is _not_ found through DI. `BodyModelBinderProvider` creates instances of this class and so your new constructor will never be used in product code.

Don't add this dependency or create the new `DefaultLoggerFactory`. Instead add another `BodyModelBinderProvider` constructor that includes an `ILoggerFactory` parameter, change the `MvcCoreMvcOptionsSetup` constructor to add the same parameter, and wire the factory through.